### DISCUSSION TITLE
Why is there no public folder and when I create it, it's in the gitignore?

### DISCUSSION THREAD

**Discussion by jorisfalter** [gethinode/hinode#290] [Help] [answered]
Apologies for abusing the issue channel as a customer service. 

I'm trying to deploy Hinode on Github, but I keep seeing the "readme" content when opening the page. 

I followed [this solution ](https://stackoverflow.com/questions/66421933/hugo-website-not-rendering-in-github-pages) to create a public folder but I assume I'm working against how you intended it to be used?

Many thanks

**Comment by markdumay** [comment]
I haven't deployed the theme to GitHub pages myself yet, mostly because it does not support configuration of security headers. However, have you tried following Hugo's [deployment instructions](https://gohugo.io/hosting-and-deployment/hosting-on-github/)?

A key difference with the main Hugo site is that Hinode uses npm. The correct build command is `npm run build`. GitHub provides a [starter template](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs).

I'm not sure how familiar you are with GitHub actions. Let me know if you need any further help.

**Comment by jorisfalter** [comment]
Thanks for the reply, I'm pretty much a beginner in Hugo and a self-taught dev overall. 

I tried Netlify instead of github, but in that case it always redirects to the demo page https://hinode-demo.markdumay.org/

I wanted to use the theme for a simple blog with some content and pics and figured I could just deploy and start modifying but deploying already proves harder than I thought.

**Comment by markdumay** [comment]
No worries, Hugo can be pretty intimidating in the beginning. You should adjust `baseURL` in the file `config/default/config.toml` too. It probably still points to the demo page of Hinode. You could share your repository if you're still stuck, so I can have a look at it.

I'm working on a more elaborate documentation, but it's still work in progress.

**Comment by jorisfalter** [comment]
Thanks!

[This is the Github repository](https://github.com/jorisfalter/HinodeTest) I'm using, it's literally a clone from the original, except that I changed `baseURL` to `https://magical-torrone-3d2095.netlify.app`

It seems to break the layout when I deploy to [Netlify](https://magical-torrone-3d2095.netlify.app)

Edit: Note, it appends `en` to the URL. When I manually remove the `en` it redirects to the demo site.

**Comment by markdumay** [comment]
I noticed you used the main repository as starting point. Unless you plan to do a lot of customization, I suggest you use the [child theme](https://github.com/markdumay/hugo-theme-hinode-child) instead. There's button called `Use this template` to instantiate a repository. The child theme uses a npm reference to the main theme of Hinode, and has far less source files in its repository. It will keep your personal repository more tidy.

Regarding the main repository, the main branch is unstable right now. I'm fixing quite a few issues, but the current version is still in alpha. If you prefer to use the main repository instead of the child repository, I suggest you use the latest stable branch:

```bash
git clone --depth 1 --branch v0.7.9 https://github.com/markdumay/hugo-theme-hinode.git
cd hugo-theme-hinode
npm install
npm run start
```

**Comment by markdumay** [comment, marked as answer]
I added documentation on how to deploy to [Netlify](https://hinode-demo.markdumay.org/en/docs/0.8/getting-started/hosting-and-deployment/#host-on-netlify). Hope this helps! I'm closing the issue for now, let me know if you need additional assistance.

**Comment by jorisfalter** [comment]
Cool, thanks a lot for your help!
